design/open/gc-two-steps-separation.md

```diff
@@ -0,0 +1,71 @@
+# Separate GC into two separately runnable steps
+
+## Motivation
+
+Improve the resilience of GC (separation of concerns, pure functions, and traceability) and allow examination
+of intermediate results.
+
+## How?
+
+The `lakefs.debug.gc.no_delete` flag lets users run GC and generate intermediate results,
+yet GC doesn't continue over those results, and retrieving them manually is quite tedious.  
+Both problems can be solved by introducing two flags and changing the `lakefs.debug.gc.no_delete`:
+
+1. `lakefs.debug.gc.no_delete` -> `lakefs.gc.only_mark` (Boolean): This flag instructs the GC operation to only mark, i.e. collect, the addresses that are intended to be deleted.
+2. `lakefs.gc.only_sweep` (Boolean): The GC operation will only sweep, i.e. delete, the marked addresses based on the provided mark ID.
+3. `lakefs.gc.mark_id` (String): This flag specifies which ID the GC will use to generate intermediate results and write outputs to.
```

Comment:
@ozkatz had a great idea to use Terraform terminology here: 
only mark -> plan 
only sweep -> apply with a plan id 
mark id -> plan id

@Jonathan-Rosenberg WDYT?

Reply:
I actually think that we should GC terminology with the GC...

Reply:
Ok, I get it. makes sense

Reply:
I prefer mark/sweep which is established terminology for this.

Neither mark/sweep nor plan/apply make it entirely clear that the second stage will delete stuff.  But sweep is clearer, that's another point in its favour.